clamp/clamp.tex

```diff
@@ -0,0 +1,331 @@
+
+%\begin{lemma}[Soundness of the stability relation]\label{lemma:stability}
+% Give it a name
+%For every $(\din, \dout)$ pair, the function $\Relation(\din, \dout)$ returns $\False$ whenever there exists some $x$ such that $d_{\X}(x, x') \leq \din$ and $d_{\Y}(T(x), T(x')) > \dout$,
+%where $T$ denotes the clamping transformation. %and $\Relation(\din, \dout)$ the clamping stability relation. % defined in pseudocode xx; specify where the Relation is defined better
+%\end{lemma}
+
+% Must show that the relation fails for every $(d_{in}, d_{out})$ pair where there exists some $x$ such that $d_{\mathcal{X}}(x, x') \leq d_{in}$ and $d_{\mathcal{Y}}(T(x), T(x')) > d_{out}$.
+
+%\begin{proof}
+%Since clamping is 1-stable, $d_{\Y}(T(x), T(x')) \leq d_{\X}(x, x')$ for all $x, x' \in \X$. (This is a recursive argument with Theorem 2! I think it has to go)
+%\end{proof}
+        
+% MS EDIT1: I adjusted the statement to negate it-- the relation must fail for every pair that does not uphold the worst-case change in distances. It is fine for the relation to also fail for valid pairs-- there are many proofs out there that are not tight. Put another way, the relation need only fail everything that is not DP, it does not need to accept everything that is DP.
+
+% Should "input parameters" be more specified?
+% "appropriately quantified" sounds vague
+%\grace{Why is this called privacy guarantee and not stability something? I thought it should only be a "privacy guarantee" for measurements not transformations.}
+%\silvia{So ``privacy relation" for measurements $=$ ``stability relation" for transformations, but both are providing the DP guarantee. But maybe ``DP guarantee" is not the best name}
+
+\smallskip
+\textbf{(Domain-metric compatibility).}  For \texttt{MakeClamp}, both cases correspond to showing that \texttt{VectorDomain(T)} is compatible with symmetric distance. This follows directly from the definition of symmetric distance, as stated in \href{https://www.overleaf.com/project/60d214e390b337703d200982}{``List of definitions used in the proofs"}. The \textit{appropriate output domain property} shown above ensures that \texttt{output\_domain} is \texttt{VectorDomain(T)}, and hence also compatible with the \texttt{output\_metric} symmetric distance.
+
+\smallskip
+\textbf{(Stability guarantee).} Throughout the stability guarantee proof, we can assume that $\function(v)$ and $\function(w)$ are in the correct output domain, by the \textit{appropriate output domain property} shown above. 
+
+Since by assumption $\Relation(\din, \dout) = \True$, by the \texttt{MakeClamp} stability relation (as defined in line~\ref{line:rel} in the pseudocode), we have that $\din \leq \dout$. Moreover, $v, w$ are assumed to be $\din$-close. By the definition of the symmetric difference metric, this is equivalent to stating that $d_{Sym}(v, w) = |\MultiSet(v) \Delta \MultiSet(w)| \leq \din$.
```

Comment:
should add the definition of the histogram notation to the proof_defns file and refer to that rather than the OpenDP programming framework paper

Reply:
The definition of histogram notation is already part of the proof_defns file (see Definitions 2.4 and 2.5); I was citing the programming framework for completeness, but I will take it out.